move/ping-pong/sources/ping_pong.move

```diff
@@ -128,203 +161,304 @@ module ping_pong::ibc {
         };
     }
 
-    public entry fun channel_open_try(
-        channel_state: u8,
-        channel_order: u8,
-        connection_id: u32,
-        counterparty_channel_id: u32,
-        version: vector<u8>,
-        counterparty_version: vector<u8>,
-        proof_init: vector<u8>,
-        proof_height: u64
-    ) acquires PingPong, SignerRef {
-        // TODO(aeryz): save the channel here
-        ibc::channel_open_try(
-            &get_signer(),
-            get_self_address(),
-            channel_state,
-            channel_order,
-            connection_id,
-            counterparty_channel_id,
-            version,
-            counterparty_version,
-            proof_init,
-            proof_height
-        );
-
+    public fun on_channel_open_try(
+        _ordering: u8,
+        _connection_id: u32,
+        _channel_id: u32,
+        _counterparty_channel_id: u32,
+        _version: vector<u8>,
+        _counterparty_version: vector<u8>
+    ) acquires PingPong {
         if (borrow_global<PingPong>(get_vault_addr()).channel_id != 0) {
             abort ERR_ONLY_ONE_CHANNEL
         };
     }
 
-    public entry fun channel_open_ack(
-        channel_id: u32,
-        counterparty_version: vector<u8>,
-        counterparty_channel_id: u32,
-        proof_try: vector<u8>,
-        proof_height: u64
-    ) acquires PingPong, SignerRef {
-        // Store the channel_id
-        ibc::channel_open_ack(
-            &get_signer(),
-            get_self_address(),
-            channel_id,
-            counterparty_version,
-            counterparty_channel_id,
-            proof_try,
-            proof_height
-        );
+    public fun on_channel_open_ack(
+        channel_id: u32, _counterparty_channel_id: u32, _counterparty_version: vector<u8>
+    ) acquires PingPong {
         borrow_global_mut<PingPong>(get_vault_addr()).channel_id = channel_id;
     }
 
-    public entry fun channel_open_confirm(
-        channel_id: u32, proof_ack: vector<u8>, proof_height: u64
-    ) acquires PingPong, SignerRef {
-        ibc::channel_open_confirm(
-            &get_signer(),
-            get_self_address(),
-            channel_id,
-            proof_ack,
-            proof_height
-        );
-
+    public fun on_channel_open_confirm(channel_id: u32) acquires PingPong {
         borrow_global_mut<PingPong>(get_vault_addr()).channel_id = channel_id;
     }
 
-    public entry fun channel_close_init(
-        _port_id: String, _channel_id: String
-    ) {
+    public fun on_channel_close_init(channel_id: u32) {
         abort ERR_INFINITE_GAME
     }
 
-    public entry fun channel_close_confirm(
-        _port_id: String, _channel_id: String
-    ) {
+    public fun on_channel_close_confirm(channel_id: u32) {
         abort ERR_INFINITE_GAME
     }
 
-    #[view]
-    public fun get_vault_addr(): address {
-        object::create_object_address(&@ping_pong, IBC_APP_SEED)
+    fun on_recv_packet_deconstruct(
```

Comment:
it would be so nice if we are able to extract these into a separate module or some kind of a helper library. That way users will be able to plug and go.